ChaosMod/Effects/db/Misc/MiscMajorTurbulence.cpp

```diff
@@ -0,0 +1,79 @@
+#include <stdafx.h>
+
+static Hash miljethash;
+
+static Hash pilothash;
+static Vehicle veh;
+
+static int spawnTimer = -1;
+static int currenttick = 1;
+
+static int offsetx;
+static int offsety;
+
+static void onStart() {
+	miljethash = GET_HASH_KEY("luxor");
+	pilothash = GET_HASH_KEY("s_m_m_pilot_01");
+	
+	offsetx = g_Random.GetRandomInt(-100, 100);
+	offsety = g_Random.GetRandomInt(-100, 100);
+}
+
+
+static void OnTick()
+{
+
+	LoadModel(miljethash);
+	LoadModel(pilothash);
+	int current_time = GET_GAME_TIMER();
+	
+	currenttick++;
+	if (current_time > spawnTimer + 4000) {
+		spawnTimer = current_time;
+		Vector3 playerPos = GET_ENTITY_COORDS(PLAYER_PED_ID(), true);
+		offsetx = g_Random.GetRandomInt(-85, 85);
+		offsety = g_Random.GetRandomInt(-85, 85);
+		switch (g_Random.GetRandomInt(0, 4)) {
+			case 0:
+				veh = CreatePoolVehicle(miljethash, playerPos.x + offsetx, playerPos.y - 400 + offsety, playerPos.z + 400, 0.0f);
+				break;
+			case 1:
+				veh = CreatePoolVehicle(miljethash, playerPos.x + offsetx, playerPos.y + 400 + offsety, playerPos.z + 400, 180.0f);
+				break;
+			case 2:
+				veh = CreatePoolVehicle(miljethash, playerPos.x - 400 + offsetx, playerPos.y + offsety, playerPos.z + 400, 270.0f);
+				break;
+			case 3:
+				veh = CreatePoolVehicle(miljethash, playerPos.x - 400 + offsetx, playerPos.y + offsety, playerPos.z + 400, 90.0f);
+				break;
+			default:
+				veh = CreatePoolVehicle(miljethash, playerPos.x + offsetx, playerPos.y - 400 + offsety, playerPos.z + 400, 0.0f);
+				break;
+		}
```

Comment:
Since everything just seems to have a +/- 400 offset, this could be drastically simplified. Something like
```cpp
...
int variant = g_Random.GetRandomInt(0, 4);
float rotation = 90.0f * variant;
if (variant % 2 === 0) {
  offsetX -= 400;
} else {
  offsetY -= 400;
}

veh = CreatePoolVehicle(miljethash, playerPos.x + offsetX, playerPos.y + offsetY, playerPos.z + 400, rotation);
```